src/DetailsView/components/failure-instance-panel-control.tsx

```diff
@@ -88,6 +97,62 @@ export class FailureInstancePanelControl extends React.Component<FailureInstance
             closeButtonAriaLabel: null,
         };
 
+        if (this.props.featureFlagStoreData && this.props.featureFlagStoreData[FeatureFlags.manualInstanceDetails]) {
+            return this.panelWithSelectorAndSnippet(testStepConfig, panelProps);
+        } else {
+            return this.panelWithoutSelectorAndSnippet(testStepConfig, panelProps);
+        }
+    }
+
+    protected panelWithSelectorAndSnippet = (testStepConfig: Readonly<Requirement>, panelProps: GenericPanelProps): JSX.Element => {
+        return (
+            <GenericPanel {...panelProps}>
+                {testStepConfig.addFailureInstruction}
+                <a className="learn-more"> Learn more about adding failure instances </a>
+                <TextField
+                    className="selector-failure-textfield"
+                    label="CSS Selector"
+                    multiline={true}
+                    rows={8}
+                    value={this.state.selector}
+                    onChange={this.onSelectorChange}
+                    resizable={false}
+                    defaultValue="CSS selector"
+                />
+                <div>Note: If the CSS selector maps to multiple snippets, the first will be selected.</div>
+                <div>
+                    <DefaultButton text="Validate CSS selector" onClick={this.onValidateSelector} disabled={this.state.selector === ''} />
+                </div>
+                <div>
+                    <label>Code Snippet</label>
+                    <div>{this.state.snippet}</div>
+                </div>
+                <TextField
+                    className="observed-failure-textfield"
+                    label="Observed failure"
```

Comment:
I've left it as Observed Failure for now just because that's what it currently is online and since I'm using the Flagged Component I didn't want to change the UI until the feature is released entirely. Let me know if you think I should still change now!

Reply:
I understand your argument here and I may be fine with it. I just see on small issue with it, the general idea of the using feature flag to hide "under development" feature is to make sure we can develop the new UI/UX without altering the "old" one. By not updating the usage of this`<TextField>` (which contains the `"Observed failure"` text) we're not fully following that design principle. Now, I don't think is a big issue, it's just, any time we demo this we need to make a note for the audience that this text is "fixed" and not behind the feature flag, which, again, I guess it's fine but it's kind of weird. If you really want to be crisp about the feature flag use, it seems better to move this behind the feature flag too. 
Up to you at this point.